### PR TITLE
feat(UI): Attempt on displaying an error when sourcing wrong files

### LINKED ISSUE #49492: Show warning to users when they try to source the wrong files
### Is your feature request related to a problem? Please describe.

When a user tries to source `style.css` (rather than `styles.css`) this fails silently.  Similarly, sourcing anything other than `script.js` will result in a rather cryptic error: `Uncaught SyntaxError: Unexpected token '<'`.

This is needlessly frustrating.

### Describe the solution you'd like

The console should show a helpful warning. For example

> You have tried to source style.css, but that does not exist. The only files that can be sourced are styles.css and script.js.

### Describe alternatives you've considered

Nothing comes to mind.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by CallmeHongmaybe**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #49492

### CODE REVIEW

**Comment by CallmeHongmaybe** [comment]
I've closed the original PR since it's getting messier over there. And upon working on your suggestions I think that it works better now, but one thing remains to be answered is that `updatePreviewSaga` somehow runs twice, which causes the error messages to appear twice as well. 

Screen Shot 2023-07-29 at 14 13 32

**Comment by ojeytonwilliams** [comment]
Sorry for the delay, @CallmeHongmaybe, I'm a little swamped at the moment. It does look like your code is working well, though!

> updatePreviewSaga somehow runs twice, which causes the error messages to appear twice as well.

That's something that will need to be address before we can get this PR in, even though it's a separate issue. I'll try and get that fixed soon so we can move forward with this.

In the mean time, if you have a chance to write tests that confirm that only one 404 message is shown that would be a huge help. Since, clearly, this is a unexpected interaction, it's best we have tests for it.

**Comment by CallmeHongmaybe** [comment]
I think it's all good at this point.

**Comment by ojeytonwilliams** [comment]
@raisedadead until the `updatePreviewSaga` is fixed, this can't go in. Also, we do need tests to confirm that the messages are only being shown once.

**Comment by Sembauke** [comment]
@ojeytonwilliams I think that pr is in right?

**Comment by ojeytonwilliams** [comment]
Not to my knowledge, @Sembauke. Which PR are you thinking of?

File: client/src/templates/Challenges/utils/frame.ts
Comment (naomi-lgbt) [inline client/src/templates/Challenges/utils/frame.ts:312]:
```suggestion
            return `${urlAddr} does not exist. Only files that can be sourced are styles.css, script.js, or remote files`;
```

**Comment by raisedadead** [comment]
Hey @ojeytonwilliams what's the most productive way of moving this forward? I can't seem to make any sense of the thread above 😓 ?

**Comment by ojeytonwilliams** [comment]
The issue is that these warnings are shown twice.

If memory serves, we need a couple of things.

The `updatePreviewSaga` gets triggered twice and so the message appears twice. At least that's what I think is wrong. So, either that needs to not happen or this code needs to handle it.

The other thing is that we need a test (playwright, maybe?) so we don't accidentally start firing off multiple messages again.

**Comment by raisedadead** [comment]
Thanks for clarifying.

@CallmeHongmaybe if you are around, please add a test to lock the feature in.


@ojeytonwilliams Do you know if there ise a dedicated thread about the `updatePreview`saga? Who is the best person to investigate and keep this going forward, OR do you have pointers so I can report an issue?

It's a shame that this PR is sitting here and isn't getting maintainer attention, and I would like to unblock progress.

**Comment by ojeytonwilliams** [comment]
> Do you know if there ise a dedicated thread about the updatePreviewsaga? Who is the best person to investigate and keep this going forward, OR do you have pointers so I can report an issue?

It's a tricky one to give pointers on, but I've taken a look and I think I have a fix.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: client/src/templates/Challenges/utils/frame.ts
Comment (CallmeHongmaybe) [reply client/src/templates/Challenges/utils/frame.ts:320]: Alrighty.

**Comment by Sembauke** [comment]
Hey @CallmeHongmaybe 

Thank you for working on this, it is working great. We want to make sure your code works by adding a test sourcing the wrong files in the editor. We recently switched from Cypress to Playwright. There should be enough examples in the repository.

If you need any help writing tests, let me know.

Have a nice day and happy coding :)

**Review by raisedadead — APPROVED** [review]

**Pull request closed without merging**
